### ISSUE TITLE
Automatically save state on exit

### ISSUE THREAD

**Issue by MisterTea** [hrydgard/ppsspp#8661]
There is an option to load the most recent save upon starting PPSSPP. Can we also have an option to automatically save upon exit?  This is especially important on Android where PPSSPP is often killed because of memory pressure. It's possible to handle the kill request on Android and do a save state so progress in the game is not lost.

**Comment by vsub** [comment]
If ppsspp is not closed manually,don't expect it to do something when you close it.
It's like wanting to save a text file but kill the notepad process first.

Such feature will often either do nothing or corrupt save states

**Comment by MisterTea** [comment]
With Android, one can monitor for the app going into the background and
save state when that happens. I believe the Android memory manager also
alerts an app prior to killing it so one can do a save state then.

Even if the feature only worked on manual exits, that would still be a big
win.

In case the app is killed for another reason (e.g. the phone runs out of
battery), I agree that trying to fire off a save state is dangerous. That's
not what this issue is about.

**Comment by unknownbrackets** [comment]
The operation of saving state requires allocating memory - often a decent chunk of it.

Therefore, if the app is being closed due to memory pressure, it's probably not going to be possible to save state.  I would assume allocating a large chunk of memory would fail during a memory pressure shutdown.

-[Unknown]

**Comment by MisterTea** [comment]
Ios and Android account for this and raise the high watermark for apps that
are about to be killed to help them exit gracefully.

Regardless, it may be better to simply save state every time the app goes
into the background and nor have to worry about trying to save one just
before being killed.  On a phone, there are constant interruptions and I
have lost hours of progress because PPSSPP has been killed and I forgot to
save state.

**Comment by Benhgift** [comment]
Greetings from 2 years into the future. I'd love this feature and would be willing to work on it

**Comment by hrydgard** [comment]
Pull requests accepted :)

**Comment by unknownbrackets** [comment]
Just a note: if a save state fails to load, it restarts emulation instead - make sure not to endlessly loop in this case.

-[Unknown]

**Comment by Gamemulatorer** [comment]
Aethersx2 and Duckstation have already this but still good for PPSSPP?

**Comment by hrydgard** [comment]
Yeah could be a good idea. Maybe have a special slot for this, instead of say saving into the last uses one, which could be annoying.

**Comment by slrgt** [comment]
this would be a great feature

**Comment by alexmozaidze** [comment]
I often find myself instinctively closing the app normally without saving, which has nothing to do with memory. Lemuroid auto-saves the state into its own hidden quicksave state that is auto-loaded when a game is opened. Not having this feature led me to losing my progress in a game :(

It would be cool to see it implemented as a quality of life feature.

**Comment by gamersfan** [comment]
Any news?

**Comment by reecespieces** [comment]
This would be a great feature to have. I like to suspend and close emulator apps and have it save automatically. 

Duckstation and AetherSX2 both have this and it's great for Android.
